xmind/tests/test_workbook_saver.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from . import base
+from unittest.mock import patch, MagicMock, mock_open
+from xmind.core.saver import WorkbookSaver
+from xmind.core.const import CONTENT_XML, XMIND_EXT
+
+
+class WorkbookSaverTest(base.Base):
+
+    def setUp(self):
+        super(WorkbookSaverTest, self).setUp()
+        self._workbook = MagicMock()
+        self._test_el = WorkbookSaver(self._workbook)
+
+    def getLogger(self):
+        if not getattr(self, '_logger', None):
+            self._logger = logging.getLogger('WoorkbookSaverTest')
+        return self._logger
+
+    def test_excessive_parameters(self):
+        # _workbook = MagicMock()
+        # _element = WorkbookSaver(_workbook)
+        _parameters = [
+            ('_get_content', 0),
+          #  ('save', 1)
+        ]
+
+        with self.assertRaises(TypeError) as _ex:
+            self._test_el.save(12,13)
+
+        self.getLogger().debug("Got an exception: '%s'", _ex.exception.args[0])
+        self.assertTrue(_ex.exception.args[0].find(
+            "%s() takes" % 'save') != -1)
+
+        for pair in _parameters:
+            with self.subTest(pair=pair):
+                self._test_method_by_excessive_parameters(pair, self._test_el)
+
+    def test_WoorkbookSaver_inint(self):
+        # Call without parameter:
+        with self.assertRaises(TypeError) as _ex1:
+            WorkbookSaver()
+        self.assertTrue(_ex1.exception.args[0].find(
+                "__init__() missing") != -1)
+
+        # Call with excessive parameters
+        with self.assertRaises(TypeError) as _ex2:
+            WorkbookSaver(1,2)
+        self.assertTrue(_ex2.exception.args[0].find(
+                "__init__() takes") != -1)
+
+        # Call with one parameter
+        _test_object = WorkbookSaver('workbook')
+        self.assertEqual(_test_object._workbook,'workbook')
+
+    def test_get_content(self):
+
+        with patch('xmind.utils.join_path') as _utils_join_path:
```

Comment:
It's very difficult to read so long hierarchy of ```with``` 
I would recommend to go with patching of each method before actual call of it.

Reply:
Done